### PR TITLE
Handle parsing errors

### PR DESCRIPTION
**Pull request by jakobandersen**
(This depends on / includes #698)

In a few places the C++ domain in Sphinx is used to parse fragments, but potential errors are not handled. This tries to fix that.
One case is the parsing of the function parameters in ``doxygenfunction``, so ``.. doxygenfunction:: f(`` would crash Breathe.
The second case is parsing the function parameters in the XML, I don't know how to trigger that directly, but it should be handled now.
The third case is in the hax to insert the name in template parameters (#681). For parameters ``typename ...Args`` it will trigger a parsing error (observed in the pybind11 project).

Note, the third case is only enabled when using Sphinx 4.1 which is still not released, but probably will soon'ish.

### CODE REVIEW

**Comment by kelson42** [comment]
Any chance this PR will be reviewed? merged? released?

File: breathe/renderer/sphinxrenderer.py
Comment (bmerry) [inline breathe/renderer/sphinxrenderer.py:2052]:
```suggestion
                        # we really should use _parse_template_parameter()
```

Comment (jakobandersen) [reply breathe/renderer/sphinxrenderer.py]: Thanks

**Comment by jakobandersen** [comment]
@michaeljones, unless you object, I'll merge this later today or tomorrow.

**Review by michaeljones — APPROVED** [review]
Look good :)

**Comment by 0x8000-0000** [comment]
Thank you @michaeljones  and @jakobandersen !

Can we have a release now, please?

**Comment by michaeljones** [comment]
@vermeeren is the release manager at the moment and has been a little busy. I'm not sure when he might be able to do it. It seems reasonable to release though.

**Comment by vermeeren** [comment]
@michaeljones @0x8000-0000 I can definitely make a new release when/if that is desired. Past months there have been some irl issues causing an unfortunate lack of activity from my end, I really do appreciate everyone's work on this (including also @jakobandersen of course).

Definitely ok to ping me when it's time, perhaps after #735 is merged to avoid conflicts? I'll at the minimum finalise the changelog and make a formal release, not yet sure when personal issues are resolved fully after which I can get back to more proper maintainership. Thanks!

**Comment by michaeljones** [comment]
Thank you for the reply @vermeeren and for sharing about your current situation. I hope you're doing ok and that the personal issues are manageable and go as well as can be. 

I'm happy to merge #735. It is a bit opinionated and I feel a bit out of place having wandered back in after such a time away but I think it has merit if only a small amount. I can merge in a day or two if no one has objections.

@jakobandersen - do you have anything else you would like to see merged for a release?

**Comment by jakobandersen** [comment]
> @jakobandersen - do you have anything else you would like to see merged for a release?

I think merging #735 and then updating with whatever actually changed since the last release is the only thing I think is needed before a release. Unfortunately I haven't had a habit of adding changelog entries so I'm not sure what is missing.

**Comment by michaeljones** [comment]
I think this is the syntax for finding merges since the approximate time of the last release: https://github.com/michaeljones/breathe/pulls?q=is%3Apr+merged%3A%3E2021-05-05+

I'm happy to look through them and dig some stuff out.

We could add a PR template with a checklist for updating the CHANGELOG if desirable.

**Comment by michaeljones** [comment]
I've created PR with changelog & minor readme changes: https://github.com/michaeljones/breathe/pull/739

Hopefully when that is merged we'll be in a good place to release?

**Comment by hidmic** [comment]
I'd love to know how far are we from a release.

**Comment by michaeljones** [comment]
@vermeeren - I think we're in a place to release now if you have the time at some point?

**Comment by vermeeren** [comment]
@michaeljones The job is done, apologies for delay. Did some minor changelog update and also fixup CICD quick in #741.

Thanks to all for the hard work!

**Comment by michaeljones** [comment]
Thank you @vermeeren. Kind of you to take the time when you've got other things going on.

**Comment by 0x8000-0000** [comment]
Thank you @vermeeren and @michaeljones !

File: breathe/renderer/sphinxrenderer.py
Comment (2bndy5) [reply breathe/renderer/sphinxrenderer.py]: this is not a typo from breathe's end.

https://github.com/sphinx-doc/sphinx/blob/7d7c59aaf22f110bfc7ed5a1385e6865ccf327fa/sphinx/domains/cpp.py#L6544

**Pull request merged**
